### PR TITLE
refactor: convert handle-scroll to typescript

### PR DESCRIPTION
**Pull request by LiKang6688**

### CODE REVIEW

File: src/table/utils/handle-scroll.ts [hunk 1]
```diff
@@ -35,18 +35,18 @@ export const handleNavigateTop = (cellCoord, rootElement) => {
   if (x < MIN_ROW_COUNT) {
     tableContainer.scrollTo({
       top: 0,
-      behavior: 'instant',
+      behavior: 'auto',
```
Comment (haxxmaxx) [inline src/table/utils/handle-scroll.ts:38]: how does the behavior differ here?

File: src/table/utils/handle-scroll.ts [hunk 2]
```diff
@@ -35,18 +35,18 @@ export const handleNavigateTop = (cellCoord, rootElement) => {
...
     });
   } else {
-    const tableHead = rootElement.getElementsByClassName('sn-table-head-cell')[0];
+    const tableHead = rootElement.getElementsByClassName('sn-table-head-cell')[0] as HTMLElement;
     const rowElements = rootElement.getElementsByClassName('sn-table-row');
-    const cell = rowElements[x]?.getElementsByClassName('sn-table-cell')[y];
+    const cell = rowElements[x]?.getElementsByClassName('sn-table-cell')[y] as HTMLElement;
```
Comment (haxxmaxx) [inline src/table/utils/handle-scroll.ts:43]: the sanity check here is a bit strange. There is a `?.` and then assume that cell exist in the if block below. Is there a case where cell is undefined?

File: src/table/utils/selections-utils.ts [hunk 3]
```diff
@@ -35,7 +35,7 @@ type AddSelectionListenersArgs = {
   selectionDispatch: React.Dispatch<Action>;
   setShouldRefocus(): void;
   keyboard: stardust.Keyboard;
-  tableWrapperRef: React.MutableRefObject<HTMLElement>;
+  tableWrapperRef: React.MutableRefObject<HTMLDivElement>;
```
Comment (haxxmaxx) [inline src/table/utils/selections-utils.ts:38]: 👍

File: src/table/utils/handle-scroll.ts (hunk 1 above)
Comment (LiKang6688) [reply src/table/utils/handle-scroll.ts:38]: no difference. 'instant' is just not available in typescript
Screenshot 2022-08-12 at 11 11 30

Comment (haxxmaxx) [reply src/table/utils/handle-scroll.ts:38]: ok

File: src/table/utils/handle-scroll.ts (hunk 2 above)
Comment (LiKang6688) [reply src/table/utils/handle-scroll.ts:43]: rowElements = [rootElement,  rootElement];
rowElements[2] is undefined
undefiiend.?getElementsByClassName('sn-table-cell')[y] is undefined

Gonna fix

**Review by haxxmaxx — APPROVED** [review]

**Pull request merged**